homeworks/ihor.zapisochniy_IgorFromUA/GamesProject/scripts/games.js

```diff
@@ -0,0 +1,233 @@
+                                <span class="semiBold">Release_date:</span>    ${releaseDate}
+                            </li>
+                        </ul>
+                   `;
+}
+function createCardElement(game) {
+    const cardElement = document.createElement('li');
+    cardElement.classList.add('card');
+    cardElement.innerHTML = createGameCardStr(game);
+    return cardElement;
+}
+function filterGames(gamesArr) {
+    const selectedGenre = document.querySelector('[data-filter-select]').value;
+    const isPlatformGame = document.querySelector('[data-type-games="platform"]').checked;
+    const isOnlineGame = document.querySelector('[data-type-games="online"]').checked;
+    const isNewGame = document.querySelector('[data-age-games="new"]').checked;
+    const isOldGame = document.querySelector('[data-age-games="old"]').checked;
+    const searchTerm = document.querySelector('[data-filter-search]').value;
+    return gamesArr.filter((game) => {
+        const {
+            genre,
+            platform,
+            release_date: releaseDate,
+            title,
+            short_description: description,
+        } = game;
+        const releaseYear = new Date(releaseDate).getFullYear();
+        if (selectedGenre !== 'genre' && selectedGenre !== genre.toLowerCase()) return false;
+        if (isPlatformGame && platform !== 'PC (Windows)') return false;
+        if (isOnlineGame && platform !== 'Web Browser') return false;
+        if (isNewGame && !isOldGame && releaseYear < 2020) return false;
+        if (isOldGame && !isNewGame && releaseYear > 2010) return false;
+        if (isOldGame && isNewGame && releaseYear < 2020 && releaseYear > 2010) return false;
+        return !(searchTerm !== '' && !title.toLowerCase().includes(searchTerm.toLowerCase())
+            && !description.toLowerCase().includes(searchTerm.toLowerCase()));
+    });
+}
+function markedSearchText(search, originString) {
+    if (search === '') return originString;
+    const regex = new RegExp(search, 'ig');
+    return originString.replaceAll(regex, (match) => `<span class="filter-matched-text">${match}</span>`);
+}
+function renderCards(container, arrGames) {
+    container.innerHTML = '';
+    const fragment = new DocumentFragment();
+    arrGames.forEach((game) => {
+        fragment.append(createCardElement(game));
+    });
+    container.append(fragment);
+}
+function reRenderCards() {
+    const searchTerm = document.querySelector('[data-filter-search]').value;
+    const filterArrGames = filterGames(games);
+    const markedTextFilterArrGames = filterArrGames.map((game) => {
+        const copyGame = { ...game };
+        copyGame.title = markedSearchText(searchTerm, game.title);
+        copyGame.small_description = markedSearchText(searchTerm, game.small_description);
+        return copyGame;
+    });
+    renderCards(CARDS_LIST, markedTextFilterArrGames);
+}
+renderCards(CARDS_LIST, games);
```

Comment:
Hey, @IgorFromUA, thanks for a good work!
Please add an init function
> Додайте функцію init яка буде навішувати обробники подій, та відмальовувати картки без фільтації уперше, після завантаження сторінки. Це буде точка входу у ваш скріпт, те місце де все починається.

Reply:
added an init function